### PR TITLE
Adding proposer: swap lhs and rhs

### PR DESCRIPTION
**Pull request by na2021**
Pull request for the swap lhs and rhs proposers

### CODE REVIEW

File: src/trs/rewrite.rs
Comment (joshrule) [inline src/trs/rewrite.rs]: Looks like this function can be deleted: you never use it elsewhere.

File: src/trs/rewrite.rs
Comment (joshrule) [inline src/trs/rewrite.rs:384]: Can you please add a test where the function fails (i.e. returns `None`)?

File: src/trs/rewrite.rs
Comment (joshrule) [inline src/trs/rewrite.rs]: Can you please add a test here and for `swap_lhs_and_rhs_vec` to give people some idea of how they work?

File: src/trs/rewrite.rs
Comment (joshrule) [inline src/trs/rewrite.rs]: Hmm... why do you think this should be `pub`?

**Review by joshrule — CHANGES_REQUESTED** [review]
Thanks, @na2021 ! Looks good overall, but please respond to the issues raised in the comments. After these are satisfied, I'll merge.

File: src/trs/rewrite.rs
Comment (joshrule) [reply src/trs/rewrite.rs]: Great - thanks for the new test. Can you also, as with the others, add cases that result in `Err` rather than `Ok`, both here and for `swap_lhs_and_rhs_vec`?

File: src/trs/rewrite.rs
Comment (joshrule) [reply src/trs/rewrite.rs]: Hmm... still not sure if this should be `pub`.

**Review by joshrule — CHANGES_REQUESTED** [review]
Looking better. Please address the below.

File: src/trs/rewrite.rs
Comment (na2021) [reply src/trs/rewrite.rs]: So right now I have it returning the original TRS if there is an issue with the swap, should it fail and Error out or should we keep it as it is?

Comment (joshrule) [reply src/trs/rewrite.rs]: I'd prefer to see it return an `Err`. Otherwise, we don't need to return a `Result`, but `Result` provides useful information: did what I was trying to do (i.e. generate a new `TRS`) succeed or fail? That gets lost if you silently return the original and pretend that it's somehow new.

Comment (na2021) [reply src/trs/rewrite.rs]: Should I create a new Err type for this specific error or should I just use a pre - existing one such as the options exhausted variant?

Comment (joshrule) [reply src/trs/rewrite.rs]: options exhausted is a fine choice.

**Review by joshrule — APPROVED** [review]

**Comment by joshrule** [comment]
I'll merge once I finish with the outstanding PR upstream. It would be great if you could push updated documentation for `swap_lhs_and_rhs_vec` and `swap_lhs_and_rhs` that also includes a test resulting in an `Err`, but I won't let that hold up the merge.

**Pull request merged**
